src/GafferOSL/OSLObject.cpp

```diff
@@ -540,7 +521,7 @@ void OSLObject::updatePrimitiveVariables()
 			oslCode()->parametersPlug()->addChild( codeClosurePlug );
 			codeClosurePlug->setInput( valuePlug );
 
-			code += prefix + "out += " + codeClosurePlug->getName().string() + ";\n";
+			code += prefix + "out = out + " + codeClosurePlug->getName().string() + ";\n";
```

Comment:
just curious, what is this change fixing?

Reply:
It seemed that OSL was totally blind to `+=` when it came to figuring out `closures_needed`. There's a similar oddity whereby we need to accumulate into `out` and only assign to `Ci` in the final step as well. I didn't have time to look into the OSL source to see what was up.